### DISCUSSION TITLE
Is there an ESLint validator?

### DISCUSSION THREAD

**Discussion by binarykitchen** [eslint/eslint#18832] [General]
Yo!

For months, we developed a React app and only realised today that ESLint was broken due to a silly mistake:

```js
module.exports = {
  root: true,
  parser: "@typescript-eslint/parser",
  ...
  // Configure ESLint for Frontend (React)
  overrides: [
    {
      files: ["./src/client/"], // <---------------- this is the mistake
      env: {
        browser: true,
      },
      extends: [
        "plugin:react/recommended",
        "plugin:react-hooks/recommended",
        "plugin:react/jsx-runtime",
        "plugin:import/react",
      ],            
      ...
```

That files override was just pointing to a directory. It didn't complain about the React mistakes.

Today I changed to - using wildcards - to `files: ["./src/client/**"]` and booom, suddenly it worked. Finally our React-related mistakes on the browser side were reported, not before.

That said, I wonder why ESLint didn't complain about it. Does it perform some validation toward invalid ESLint configurations itself?

Using ESLint version `8.57.0` here, along with Node.js v 20.16

What do you think of that? Shouldn't ESLint validate this?

Thanks, M.

**Comment by error-four-o-four** [comment]
hey there,

one could use the cli flag [print-config](https://eslint.org/docs/latest/use/command-line-interface#--print-config) on a specific file to see if it matches the defined config.

Otherwise I'd agree. Eslint could print a warning like e.g. prettier does.

But I also vaguely remember some other kind of validation. I'll have to search ...

**Reply by binarykitchen** [reply]
Thanks. `print-config` is new to me. Noted.

But then, I bet, ...

... if ESLint prints warnings about wrong configurations, this will be a huge improvement for everyone and the community.

**Comment by error-four-o-four** [comment]
actually there is some validation here https://github.com/eslint/eslint/blob/main/lib%2Fcli-engine%2Ffile-enumerator.js#L178

**Comment by binarykitchen** [comment]
@error-four-o-four Then why is this `NoFilesFoundError` not thrown when I run `eslint` with a wrong configuration?

Or does this happen with the newer ESLint version only?

**Comment by error-four-o-four** [comment]
By default eslint searches for `.js` files within the defined directory (see [here](https://github.com/eslint/eslint/blob/main/lib%2Fcli-engine%2Ffile-enumerator.js#L245))

In case that your directory `src/clients/` contains js files I'd argue that it's working as intended.

please consider: I'm not a member of the eslint team. All my assumptions are based upon my understanding of the eslint codebase. Please correct me if I'm wrong

**Comment by binarykitchen** [comment]
@error-four-o-four Ah, this explains since all my files were in TypeScript. How about changing

```js
        // If `--ext` option is present, use it.
        if (extensionRegExp) {
            return extensionRegExp.test(filePath);
        }

        // `.js` file is target by default.
        if (filePath.endsWith(".js")) {
            return true;
        }
```

to honouring anything with `extensionRegExp` as well? Like

```js
        // If `--ext` option is present, use it.
        if (extensionRegExp) {
            return extensionRegExp.test(filePath);
        }

        if (fextensionRegExp.test(filePath)) {
            return true;
        }
```

(Just an idea, I do not know the whole code nor have tested this yet)

**Reply by error-four-o-four** [reply]
well

There's a discussion about typescript support in future versions but this hasn't been settled yet. We'll have to wait.

Regarding the validation:
The advantage of eslint is that it is framework agnostic and highly configurable.

I'd argue that it would be a heavy burdon for the eslint devs if they were responsible for validating every possible extension (.jsx, .ts, .svelte, .vue, .whatever)

**Reply by binarykitchen** [reply]
True, as long as Node.js core does not support TS, let's not stress about it.

Agree, it would be a heavy burden.

All I am raising is that I got sucked into a mess for months, why nothing gets linted until I've found out why. Some warning or documentation would be good to save others from the same trouble.

**Reply by error-four-o-four** [reply]
Indeed.

So this might be an opportunity to improve the docs. https://eslint.org/docs/latest/use/configure/configuration-files#configuration-objects

**Reply by binarykitchen** [reply]
@error-four-o-four Cool. If you are a maintainer, and I'll do a Pull Request to polish the documentation about that, are you in?

**Reply by error-four-o-four** [reply]
As mentioned before I'm not a maintainer.
Feel free to ask on github or [discord](https://eslint.org/chat)

**Reply by binarykitchen** [reply]
Ok, respekt
